qiskit/transpiler/passes/layout/dense_layout.py

```diff
@@ -98,11 +98,9 @@ def run(self, dag):
 
         best_sub = self._best_subset(num_dag_qubits, num_meas, num_cx)
         layout = Layout()
-        map_iter = 0
+        for i, qubit in enumerate(dag.qubits):
+            layout.add(qubit, int(best_sub[i]))
         for qreg in dag.qregs.values():
-            for i in range(qreg.size):
-                layout[qreg[i]] = int(best_sub[map_iter])
-                map_iter += 1
             layout.add_register(qreg)
```

Comment:
I don't even know why `Layout` tracks registers, but it does seem to be part of the interface, so best to keep it.  I'm guessing it's from before, when registers owned qubits.

Reply:
Yeah, neither am I, and tbh I don't think it is 100% needed here anymore as without this all the tests passed, but I kept it in just in case there was something I wasn't considering.